src/wazuh_modules/wm_database.c

```diff
@@ -250,6 +250,38 @@ void wm_check_agents() {
     }
 }
 
+int wm_truncate_groups_file(const char *id, char *groups) {
```

Comment:
We can simplify this logic by using wstr_split(), getting the index of the correct element, and after this call to wdb_set_agent_groups